libdevcore/OverlayDB.h

```diff
@@ -50,10 +55,7 @@ class OverlayDB: public MemoryDB
 private:
 	using MemoryDB::clear;
 
-	std::shared_ptr<ldb::DB> m_db;
-
-	ldb::ReadOptions m_readOptions;
-	ldb::WriteOptions m_writeOptions;
+	std::shared_ptr<db::DatabaseFace> m_db;
```

Comment:
I don't see the point of using `shared_ptr` here, can we change it to `unique_ptr` ?

Reply:
Oh, the point is in having copy assignment I guess... Could you then add copy constructor & assignment with `=default` to make it more explicit